x/genutil/client/cli/init.go

```diff
@@ -114,12 +121,51 @@ func InitCmd(mbm module.BasicManager, defaultNodeHome string) *cobra.Command {
 			genFile := config.GenesisFile()
 			overwrite, _ := cmd.Flags().GetBool(FlagOverwrite)
 
+			stakingBondDenom, _ := cmd.Flags().GetString(FlagStakingBondDenom)
+
 			if !overwrite && tmos.FileExists(genFile) {
 				return fmt.Errorf("genesis.json file already exists: %v", genFile)
 			}
-			appState, err := json.MarshalIndent(mbm.DefaultGenesis(cdc), "", " ")
+
+			var stakingGenState map[string]json.RawMessage
+
+			if stakingBondDenom != "" {
+				stakingGenState = mbm.DefaultGenesis(cdc)
+				stakingRaw := stakingGenState[stakingtypes.ModuleName]
+				var (
+					initialStakingData simapp.GenesisState
+					finalStakingdata   simapp.GenesisState
+				)
+
+				if err := tmjson.Unmarshal(stakingRaw, &initialStakingData); err != nil {
+					return err
+				}
+
+				bz, err := tmjson.Marshal(initialStakingData)
+				if err != nil {
+					return err
+				}
+
+				stakingStr := string(bz)
+				modifiedStakingStr := strings.Replace(stakingStr, sdk.DefaultBondDenom, stakingBondDenom, 1)
```

Comment:
is there a risk that the characters `"stake"` appear somewhere else in `stakingStr`? And that we inadvertently replace something else?

Overall, I feel the correct way is to do:
```go
var stakingGenesis
clientCtx.UnmarshalJSON(stakingStr, &stakingGenesis)
stakingGenesis.Params.BondDenom = stakingBondDenom
modifiedStakingStr := clientCtx.MarshalJSON(stakingGenesis)
```

Reply:
thanks for the suggestion @amaurym, this makes sense to me